WalletWasabi.Documentation/RPC.md

```diff
@@ -0,0 +1,437 @@
+# Wasabi Remote Procedure Call interface
+
+Wasabi Wallet provides an RPC interface to interact with the wallet programmatically. The RPC server is listening by default on port 18099.
+
+## Limitations
+
+The RPC server does NOT support batch requests nor TLS communication (because it is not supported on Linux nor Mac: https://github.com/dotnet/corefx/issues/14691). Requests are served in order one by one in serie (no parallel processing).
+It is intentionally limited to serve only one whitelisted local address and it is disabled by default.
+
+# Before start (Enable RPC)
+
+The RPC server has to be configured and enabled. This is done in the `Config.json` file and the relevant settings are:
+
+* JsonRpcServerEnabled: [true | false]  (default: false)
+* JsonRpcServerPrefixes: [an array of string with prefixes]
+	(default: [	"http://127.0.0.1:18099/", "http://localhost:18099/"])
+
+# Authentication
+
+The RPC server can be configured to allow `Anonymous` access or `Basic  authentication` just by editing the
+
+* JsonRpcUser: [the username]  (default: empty)
+* JsonRpcPassword: [the password]  (default: empty)
+
+By default both `JsonRpcUser` and `JsonRpcPassword` are empty (`""`) that that means that `Anonymous` requests are allowd. On the other hand, if `JsonRpcUser` and `JsonRpcPassword` setting are not empty it means that the requester has to provide the right credential, otherwise it will get a http statuc code 401 (Unauthorized).
```

Comment:
```suggestion
By default both `JsonRpcUser` and `JsonRpcPassword` are empty `""`, which means that `Anonymous` requests are allowed. On the other hand, if `JsonRpcUser` and `JsonRpcPassword` are not empty it means that the requester has to provide the right credentials, otherwise he will get a http status code 401 (Unauthorized).
```